src/frequenz/sdk/timeseries/ev_charger_pool/_ev_charger_pool.py

```diff
@@ -62,14 +76,27 @@ def __init__(
                     component_category={ComponentCategory.EV_CHARGER}
                 )
             }
+        self._state_tracker: StateTracker | None = None
+        self._status_streams: dict[
+            int, tuple[Task[None], Broadcast[EVChargerData]]
+        ] = {}
         self._namespace: str = f"ev-charger-pool-{uuid.uuid4()}"
         self._formula_pool: FormulaEnginePool = FormulaEnginePool(
             self._namespace,
             self._channel_registry,
             self._resampler_subscription_sender,
         )
 
-    async def total_current(self) -> FormulaReceiver3Phase:
+    @property
+    def component_ids(self) -> set[int]:
```

Comment:
You probably don't want the client to mutate this, so I suggest returning a [`collections.abc.Set`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Set) instead, which is immutable.

Reply:
I'm assuming you just mean the type hint,  so that mypy would catch it when people try to mutate it.  Because the type is an ABC otherwise,  and there's no default read-only implementation.

Reply:
Yes, just the type hint.

And there is a read-only implementation, it is called `frozenset` and it is a built-in too.

Reply:
OK, the difference here is you are copying the data (and rebuilding a hash set). I guess it's not a big deal, but IMHO it would be a better balance to just make the type system enforce this.

I just tried this and for me `mypy` doesn't complain about missing `AbstractSet` but it does say that `AbstractSet` doesn't support X instead of `Set`, which is indeed a bit weird, but I'm not sure if weird enough.

```py
from __future__ import annotations

from collections.abc import Set


def f() -> Set[int]:
    return {1, 2, 3}


s1: Set[int] = f()
s2: set[int] = f()  # line 11
f().add(4)  # line 12
```

```console
$ mypy t.py
t.py:11: error: Incompatible types in assignment (expression has type "AbstractSet[int]", variable has type "Set[int]")  [assignment]
t.py:12: error: "AbstractSet[int]" has no attribute "add"  [attr-defined]
Found 2 errors in 1 file (checked 1 source file)
```

Anyway, just giving an opinion, whatever you prefer is fine with me.

Reply:
Line 11's error was confusing for me,  maybe others can figure it out,  or even better,  can just use it directly, without assigning.  

It is not a performance sensitive part, but I agree it doesn't make sense to make a copy with every call.

The other option is to use a frozenset for the internal representation,  because once created,  we don't update it internally either.

I don't know that using `abc.Set` is a good move,  but I've done it anyway,  and can leave it like this until we have more info,  and deal with it separately if necessary.